test/parallel/test-stream2-writable.js

```diff
@@ -328,7 +328,7 @@ test('dont end while writing', function(t) {
     assert(!this.writing);
     wrote = true;
     this.writing = true;
-    setTimeout(function() {
+    setImmediate(function() {
```

Comment:
Reverting to `setTimeout()` just to be 100% sure we're not modifying the test behavior in a way that might undermine the purpose of the test.